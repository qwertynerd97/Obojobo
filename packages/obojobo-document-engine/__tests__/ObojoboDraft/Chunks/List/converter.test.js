jest.mock('../../../../src/scripts/oboeditor/util/text-util')

import Converter from '../../../../ObojoboDraft/Chunks/List/converter'
const LIST_LEVEL_NODE = 'ObojoboDraft.Chunks.List.Level'

describe('List Converter', () => {
	test('slateToObo converts a Slate node to an OboNode with content', () => {
		const slateNode = {
			key: 'mockKey',
			type: 'mockType',
			data: {
				get: () => {
					return { listStyles: {} }
				}
			},
			text: 'mockText',
			nodes: [
				{
					type: LIST_LEVEL_NODE,
					data: {
						get: () => {
							return {}
						}
					},
					nodes: [
						{
							text: 'mockText',
							nodes: [
								{
									leaves: [
										{
											text: 'mockText',
											marks: [
												{
													type: 'b',
													data: {}
												}
											]
										}
									]
								}
							]
						},
						{
							type: LIST_LEVEL_NODE,
							data: {
								get: () => {
									return {}
								}
							},
							nodes: [
								{
									text: 'mockText',
									nodes: []
								}
							]
						}
					]
				}
			]
		}
		const oboNode = Converter.slateToObo(slateNode)

		expect(oboNode).toMatchSnapshot()
	})

	test('oboToSlate converts an OboNode to a Slate node', () => {
		const oboNode = {
			id: 'mockKey',
			type: 'mockType',
			content: {
				listStyles: {},
				textGroup: [
					{
						text: { value: 'mockLine1' }
					},
					{
						text: { value: 'mockLine1' },
						data: { indent: 5 }
					},
					{
						text: { value: 'mockLine2' }
					}
				]
			}
		}
		const slateNode = Converter.oboToSlate(oboNode)

		expect(slateNode).toMatchSnapshot()
	})

	test('oboToSlate converts an OboNode to a Slate node with a list style', () => {
		const oboNode = {
			id: 'mockKey',
			type: 'mockType',
			content: {
				listStyles: {
					type: 'unordered',
					indents: {}
				},
				textGroup: [
					{
						text: { value: 'mockLine1' }
					},
					{
						text: { value: 'mockLine1' },
						data: { indent: 5 }
					}
				]
			}
		}
		const slateNode = Converter.oboToSlate(oboNode)

		expect(slateNode).toMatchSnapshot()
	})
})
