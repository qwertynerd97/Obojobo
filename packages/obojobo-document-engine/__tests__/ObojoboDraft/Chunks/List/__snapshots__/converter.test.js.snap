// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Converter oboToSlate converts an OboNode to a Slate node 1`] = `
Object {
  "data": Object {
    "content": Object {
      "listStyles": Object {
        "indents": Object {},
      },
      "textGroup": Array [
        Object {
          "text": Object {
            "value": "mockLine1",
          },
        },
        Object {
          "data": Object {
            "indent": 5,
          },
          "text": Object {
            "value": "mockLine1",
          },
        },
        Object {
          "text": Object {
            "value": "mockLine2",
          },
        },
      ],
    },
  },
  "key": "mockKey",
  "nodes": Array [
    Object {
      "data": Object {
        "content": Object {
          "bulletStyle": "decimal",
          "type": undefined,
        },
      },
      "nodes": Array [
        Object {
          "nodes": Array [
            Object {
              "leaves": undefined,
              "object": "text",
            },
          ],
          "object": "block",
          "type": "ObojoboDraft.Chunks.List.Line",
        },
        Object {
          "data": Object {
            "content": Object {
              "bulletStyle": "upper-alpha",
              "type": undefined,
            },
          },
          "nodes": Array [
            Object {
              "data": Object {
                "content": Object {
                  "bulletStyle": "upper-roman",
                  "type": undefined,
                },
              },
              "nodes": Array [
                Object {
                  "data": Object {
                    "content": Object {
                      "bulletStyle": "lower-alpha",
                      "type": undefined,
                    },
                  },
                  "nodes": Array [
                    Object {
                      "data": Object {
                        "content": Object {
                          "bulletStyle": "lower-roman",
                          "type": undefined,
                        },
                      },
                      "nodes": Array [
                        Object {
                          "data": Object {
                            "content": Object {
                              "bulletStyle": undefined,
                              "type": undefined,
                            },
                          },
                          "nodes": Array [
                            Object {
                              "nodes": Array [
                                Object {
                                  "leaves": undefined,
                                  "object": "text",
                                },
                              ],
                              "object": "block",
                              "type": "ObojoboDraft.Chunks.List.Line",
                            },
                          ],
                          "object": "block",
                          "type": "ObojoboDraft.Chunks.List.Level",
                        },
                      ],
                      "object": "block",
                      "type": "ObojoboDraft.Chunks.List.Level",
                    },
                  ],
                  "object": "block",
                  "type": "ObojoboDraft.Chunks.List.Level",
                },
              ],
              "object": "block",
              "type": "ObojoboDraft.Chunks.List.Level",
            },
          ],
          "object": "block",
          "type": "ObojoboDraft.Chunks.List.Level",
        },
        Object {
          "nodes": Array [
            Object {
              "leaves": undefined,
              "object": "text",
            },
          ],
          "object": "block",
          "type": "ObojoboDraft.Chunks.List.Line",
        },
      ],
      "object": "block",
      "type": "ObojoboDraft.Chunks.List.Level",
    },
  ],
  "object": "block",
  "type": "mockType",
}
`;

exports[`List Converter oboToSlate converts an OboNode to a Slate node with a list style 1`] = `
Object {
  "data": Object {
    "content": Object {
      "listStyles": Object {
        "indents": Object {},
        "type": "unordered",
      },
      "textGroup": Array [
        Object {
          "text": Object {
            "value": "mockLine1",
          },
        },
        Object {
          "data": Object {
            "indent": 5,
          },
          "text": Object {
            "value": "mockLine1",
          },
        },
      ],
    },
  },
  "key": "mockKey",
  "nodes": Array [
    Object {
      "data": Object {
        "content": Object {
          "bulletStyle": "disc",
          "type": "unordered",
        },
      },
      "nodes": Array [
        Object {
          "nodes": Array [
            Object {
              "leaves": undefined,
              "object": "text",
            },
          ],
          "object": "block",
          "type": "ObojoboDraft.Chunks.List.Line",
        },
        Object {
          "data": Object {
            "content": Object {
              "bulletStyle": "circle",
              "type": "unordered",
            },
          },
          "nodes": Array [
            Object {
              "data": Object {
                "content": Object {
                  "bulletStyle": "square",
                  "type": "unordered",
                },
              },
              "nodes": Array [
                Object {
                  "data": Object {
                    "content": Object {
                      "bulletStyle": undefined,
                      "type": "unordered",
                    },
                  },
                  "nodes": Array [
                    Object {
                      "data": Object {
                        "content": Object {
                          "bulletStyle": undefined,
                          "type": "unordered",
                        },
                      },
                      "nodes": Array [
                        Object {
                          "data": Object {
                            "content": Object {
                              "bulletStyle": undefined,
                              "type": "unordered",
                            },
                          },
                          "nodes": Array [
                            Object {
                              "nodes": Array [
                                Object {
                                  "leaves": undefined,
                                  "object": "text",
                                },
                              ],
                              "object": "block",
                              "type": "ObojoboDraft.Chunks.List.Line",
                            },
                          ],
                          "object": "block",
                          "type": "ObojoboDraft.Chunks.List.Level",
                        },
                      ],
                      "object": "block",
                      "type": "ObojoboDraft.Chunks.List.Level",
                    },
                  ],
                  "object": "block",
                  "type": "ObojoboDraft.Chunks.List.Level",
                },
              ],
              "object": "block",
              "type": "ObojoboDraft.Chunks.List.Level",
            },
          ],
          "object": "block",
          "type": "ObojoboDraft.Chunks.List.Level",
        },
      ],
      "object": "block",
      "type": "ObojoboDraft.Chunks.List.Level",
    },
  ],
  "object": "block",
  "type": "mockType",
}
`;

exports[`List Converter slateToObo converts a Slate node to an OboNode with content 1`] = `
Object {
  "children": Array [],
  "content": Object {
    "listStyles": Object {
      "indents": Array [
        Object {},
        Object {},
      ],
    },
    "textGroup": Array [
      Object {
        "data": Object {
          "indent": 0,
        },
        "text": Object {
          "styleList": Array [],
          "value": "mockText",
        },
      },
      Object {
        "data": Object {
          "indent": 1,
        },
        "text": Object {
          "styleList": Array [],
          "value": "mockText",
        },
      },
    ],
  },
  "id": "mockKey",
  "type": "mockType",
}
`;
